collector/shared/version.go

```diff
@@ -1,42 +0,0 @@
-package collector_shared
```

Comment:
Currently there is no code that used those functions, so I removed them. The code currently doesn't need to behave differently (pending any bugs) between versions due to the struct based output, and the "if == nil" checking on each subcomponent before it is Exported to Prometheus. For eg: the WiredTiger code works perfectly on MongoDB 2.6 because the code notices it is not in the db.serverStatus() output, and therefore never exports it. I'm hoping we can continue to use this approach unless there is a blocker.